circuitx/android/src/main/kotlin/com/slack/circuitx/android/AndroidScreenAwareNavigator.kt

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) 2023 Slack Technologies, LLC
+// SPDX-License-Identifier: Apache-2.0
+package com.slack.circuitx.android
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.annotation.CheckResult
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.remember
+import com.slack.circuit.runtime.Navigator
+import com.slack.circuit.runtime.Screen
+import kotlinx.parcelize.Parcelize
+
+/**
+ * A [Screen] that refers to a launchable Android component, such as an [Intent].
+ *
+ * These screens are intercepted by the intercepting [Navigator] returned by
+ * [rememberAndroidScreenAwareNavigator] and routed to an [AndroidScreenLauncher] that can launch it
+ * natively, such as via [Context.startActivity].
+ *
+ * This is an interop layer between the [Navigator] API and the Android platform, and not something
+ * that should be used to navigate between standard Circuit screens. It should be assumed that
+ * calling [Navigator.goTo] with an [AndroidScreen] will result in exiting the current Circuit UI.
+ */
+public interface AndroidScreen : Screen
+
+/**
+ * A simple [Intent] [Screen] that wraps an [intent] and optional [options] bundle. These are passed
+ * on to [Context.startActivity].
+ *
+ * For reuse in custom [AndroidScreenLauncher] implementations, you can call [launchWith] on
+ * instances of this.
+ */
+@Parcelize
+public data class IntentScreen(val intent: Intent, val options: Bundle? = null) : AndroidScreen {
+  public fun launchWith(context: Context) {
+    context.startActivity(intent, options)
+  }
+}
+
+@Stable
+private class AndroidScreenAwareNavigator(
+  private val delegate: Navigator,
+  private val launcher: AndroidScreenLauncher,
+) : Navigator by delegate {
+  override fun goTo(screen: Screen) {
+    when (screen) {
+      is AndroidScreen -> launcher.launch(screen)
+      else -> delegate.goTo(screen)
+    }
+  }
+}
```

Comment:
I had plans to port the Slack intercepting navigator over to Circuit in the near future. I'd still like to do that as it is a more flexible solution (it allows the caller to provide multiple interceptors if needed).

Reply:
Sounds good, let's start with this